### ISSUE TITLE
Consider the use case of a PV Secondary that is updated by a dealer using a laptop that is running as a Primary

### ISSUE THREAD

**Issue by cameronmott** [uptane/obsolete-reference-implementation#33]
In the previous workshop, this functionality was indicated as a helpful milestone, for a dealer to be able to provide updates from a Primary running on a laptop (or desktop running in a vehicle bay, the essential concept is that the Primary doesn't have to be embedded, nor does it have to be assembled with the vehicle).  When considering the architecture, I noted a few concerns, please comment if you know of others or solutions to these:
1. The Secondary may not have established trust for the Primary 
2. The Secondary may not have connection information for the Primary
3. The Secondary may be serviced by multiple Primaries (different dealerships for example) 
4. The Primary may not deliver manifest information back to the Director in a timely manner

One example that highlights these concerns - 
A vehicle is taken into the local dealership for updates and the Secondary communicates with the dealer's Primary and reports an attack through the vehicle manifest. The local dealership doesn't have the resources to provide a fix and the vehicle is transferred to another (regional) dealership. The local dealership's Primary doesn't report the vehicle manifest back to the Director (maybe a bad internet connection or end of the day the system was shut down). The regional dealership's Primary doesn't receive the attack report because it has already been reported to the local dealership's Primary (perhaps this is handled already and I'm not aware of it?). With the switch between Primaries and delay in manifest delivery, this example would exercise 1, 2, 3 and 4.
Note that the attack may not appear in the original manifest, perhaps it is detected and mitigated during the update process but the local dealership is compromised and cannot complete the update.

If already addressed, please close the issue.

**Comment by JustinCappos** [comment]
Let's certainly discuss before closing the issue.

How the manifest is handled is really up to the implementers of Uptane.  If
they want to clear it each time, use an append only log, etc., it is up to
them.  To us, it is just bytes.  Perhaps this is under specifying things,
but I would rather be under than over prescriptive in this case.

However, I do not believe it makes sense for the Primary to be able to
indicate whether a manifest was received, etc.  This is really something
that the director (who consumes the content) must say.

I suppose one point that we haven't discussed is how the secondary decides
to talk with a primary.  I have been assuming that since the primary is not
really trusted, communication with anyone claiming to be a primary
(especially if they have physical access) is fine.  However, I wonder what
others think?  Trishank: Could you perhaps go through the threat model and
think about this case carefully to see if we missed anything?

Thanks,
Justin

**Comment by trishankkarthik** [comment]
> However, I do not believe it makes sense for the Primary to be able to
> indicate whether a manifest was received, etc.  This is really something
> that the director (who consumes the content) must say.

I agree. Is the thinking that the director should include metadata about what manifest it last received from each ECU?

> I suppose one point that we haven't discussed is how the secondary decides
> to talk with a primary.  I have been assuming that since the primary is not
> really trusted, communication with anyone claiming to be a primary
> (especially if they have physical access) is fine.  However, I wonder what
> others think?  Trishank: Could you perhaps go through the threat model and
> think about this case carefully to see if we missed anything?

Okay, let me think about this, and get back to you...
